compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1045,6 +1231,22 @@ private void populateFuncSignature(BLangFunction bLFunction, FunctionSignatureNo
         }
     }
 
+    private BLangXMLQName createStartEndXMLTag(XMLNameNode xmlNameNode, DiagnosticPos pos) {
+        BLangXMLQName xmlName = (BLangXMLQName) TreeBuilder.createXMLQNameNode();
+        if (xmlNameNode.kind() == SyntaxKind.XML_QUALIFIED_NAME) {
+            xmlName.localname = createIdentifier(getPosition(((XMLQualifiedNameNode) xmlNameNode).name()),
+                    ((XMLQualifiedNameNode) xmlNameNode).name().toString());
+            xmlName.prefix = createIdentifier(getPosition(((XMLQualifiedNameNode) xmlNameNode).prefix()),
+                    ((XMLQualifiedNameNode) xmlNameNode).prefix().toString());
+        } else if (xmlNameNode.kind() == SyntaxKind.XML_SIMPLE_NAME) {
+            xmlName.localname = createIdentifier(getPosition(((XMLSimpleNameNode) xmlNameNode).name()),
+                    ((XMLSimpleNameNode) xmlNameNode).name().text());
+            xmlName.prefix = createIdentifier(null, "");
+        }
```

Comment:
else, throw an exception. Then its easy to debug, if something goes wrong.

Reply:
Consider using a switch-case

Reply:
Changed